spark/src/main/java/com/netflix/iceberg/spark/source/Reader.java

```diff
@@ -269,16 +286,26 @@ private Schema lazyExpectedSchema() {
     private final Schema tableSchema;
     private final Schema expectedSchema;
     private final FileIO fileIo;
+    private final Map<String, InputFile> inputFiles;
 
     private Iterator<InternalRow> currentIterator = null;
     private Closeable currentCloseable = null;
     private InternalRow current = null;
 
-    public TaskDataReader(CombinedScanTask task, Schema tableSchema, Schema expectedSchema, FileIO fileIo) {
+    public TaskDataReader(CombinedScanTask task, Schema tableSchema, Schema expectedSchema, FileIO fileIo,
+                          EncryptionManager encryptionManager) {
       this.fileIo = fileIo;
       this.tasks = task.files().iterator();
       this.tableSchema = tableSchema;
       this.expectedSchema = expectedSchema;
+      Iterable<InputFile> decryptedFiles = encryptionManager.decrypt(Iterables.transform(task.files(),
+          fileScanTask ->
+              EncryptedFiles.encryptedInput(
+                  this.fileIo.newInputFile(fileScanTask.file().path().toString()),
+                  fileScanTask.file().keyMetadata())));
```

Comment:
Why does a `DataFile` returns a `ByteBuffer` for keyMetadata() method instead of the wrapper we created -> `com.netflix.iceberg.encryption.EncryptionKeyMetadata` ?

Reply:
I believe it's to make the `DataFile` API cleaner and less complicated, see discussion thread here: https://github.com/apache/incubator-iceberg/pull/85#discussion_r253150641